From f9f9aa9873c5851da86d33ca75e937ac022206a3 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Wed, 8 Oct 2014 09:54:55 +0200
Subject: [PATCH] Remove smb from X-KDE-Protocols lines

smb was included in the set of supported protocols since those X-KDE-Protocol
lines got introduced with 673be8e76856c6bc39f448f3374db4ae84258952 "add
X-KDE-Protocols," but as discussed in the mail thread starting at
<http://lists.freedesktop.org/archives/libreoffice/2014-September/063621.html>
"X-KDE-Protocols=...,smb,...," actual support for smb URLs in LO under KDE is
very fragile and typically requires further steps from the user (which vary
across the various LO configure options, --enable-gio vs. --enable-gnome-vfs,
and Linux distros) to work at all.

So it is probably best to not unconditionally claim smb support in those
X-KDE_Protocols lines at least in the vanilla LO sources (esp. since the TDF
builds with --enable-gnome-vfs need a really weird OOO_FORCE_DESKTOP=gnome
workaround to make it work at all, cf.
<http://lists.freedesktop.org/archives/libreoffice/2014-October/063876.html>
"Re: X-KDE-Protocols=...,smb,...").  Any Linux distros that are confident that
smb support in LO under KDE reliably works for them can patch those lines
locally.

Change-Id: I1c247d85825c96f0d74efcce3003e327a1522fa3
---
 sysui/desktop/menus/base.desktop        | 2 +-
 sysui/desktop/menus/calc.desktop        | 2 +-
 sysui/desktop/menus/draw.desktop        | 2 +-
 sysui/desktop/menus/impress.desktop     | 2 +-
 sysui/desktop/menus/math.desktop        | 2 +-
 sysui/desktop/menus/qstart.desktop      | 2 +-
 sysui/desktop/menus/startcenter.desktop | 2 +-
 sysui/desktop/menus/writer.desktop      | 2 +-
 sysui/desktop/menus/xsltfilter.desktop  | 2 +-
 9 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/sysui/desktop/menus/base.desktop b/sysui/desktop/menus/base.desktop
index 26f1c6a..76255fe 100644
--- a/sysui/desktop/menus/base.desktop
+++ b/sysui/desktop/menus/base.desktop
@@ -31,7 +31,7 @@ X-GIO-NoFuse=true
 Keywords=Data;SQL;
 InitialPreference=5
 StartupWMClass=libreoffice-base
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/calc.desktop b/sysui/desktop/menus/calc.desktop
index fca3fe3..5aa48d2 100644
--- a/sysui/desktop/menus/calc.desktop
+++ b/sysui/desktop/menus/calc.desktop
@@ -31,7 +31,7 @@ X-GIO-NoFuse=true
 Keywords=Accounting;Stats;OpenDocument Spreadsheet;Chart;Microsoft Excel;Microsoft Works;OpenOffice Calc;ods;xls;xlsx;
 InitialPreference=5
 StartupWMClass=libreoffice-calc
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/draw.desktop b/sysui/desktop/menus/draw.desktop
index ae57504..3e20e8d 100644
--- a/sysui/desktop/menus/draw.desktop
+++ b/sysui/desktop/menus/draw.desktop
@@ -31,7 +31,7 @@ X-GIO-NoFuse=true
 Keywords=Vector;Schema;Diagram;Layout;OpenDocument Graphics;Microsoft Publisher;Corel Draw;cdr;odg;svg;
 InitialPreference=5
 StartupWMClass=libreoffice-draw
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/impress.desktop b/sysui/desktop/menus/impress.desktop
index c1f6231..b419593 100644
--- a/sysui/desktop/menus/impress.desktop
+++ b/sysui/desktop/menus/impress.desktop
@@ -31,7 +31,7 @@ X-GIO-NoFuse=true
 Keywords=Slideshow;Slides;OpenDocument Presentation;Microsoft PowerPoint;Microsoft Works;OpenOffice Impress;odp;ppt;pptx;
 InitialPreference=5
 StartupWMClass=libreoffice-impress
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/math.desktop b/sysui/desktop/menus/math.desktop
index b6981f5..e437d26 100644
--- a/sysui/desktop/menus/math.desktop
+++ b/sysui/desktop/menus/math.desktop
@@ -32,7 +32,7 @@ X-GIO-NoFuse=true
 Keywords=Equation;OpenDocument Formula;Formula;odf;MathML;
 InitialPreference=5
 StartupWMClass=libreoffice-math
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/qstart.desktop b/sysui/desktop/menus/qstart.desktop
index 2669cc3..ec4e9c2 100644
--- a/sysui/desktop/menus/qstart.desktop
+++ b/sysui/desktop/menus/qstart.desktop
@@ -25,4 +25,4 @@ NoDisplay=true
 Name=%PRODUCTNAME Quickstarter
 Comment=Hook for quickstarter startup
 X-GIO-NoFuse=true
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
diff --git a/sysui/desktop/menus/startcenter.desktop b/sysui/desktop/menus/startcenter.desktop
index 3858c89..08d4a11 100644
--- a/sysui/desktop/menus/startcenter.desktop
+++ b/sysui/desktop/menus/startcenter.desktop
@@ -30,7 +30,7 @@ Comment=The office productivity suite compatible to the open and standardized OD
 StartupNotify=true
 X-GIO-NoFuse=true
 StartupWMClass=libreoffice-startcenter
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 ##Define Actions
 Actions=Writer;Calc;Impress;Draw;Base;Math;
diff --git a/sysui/desktop/menus/writer.desktop b/sysui/desktop/menus/writer.desktop
index e22a7b4..44a6421 100644
--- a/sysui/desktop/menus/writer.desktop
+++ b/sysui/desktop/menus/writer.desktop
@@ -31,7 +31,7 @@ X-GIO-NoFuse=true
 Keywords=Text;Letter;Fax;Document;OpenDocument Text;Microsoft Word;Microsoft Works;Lotus WordPro;OpenOffice Writer;CV;odt;doc;docx;rtf;
 InitialPreference=5
 StartupWMClass=libreoffice-writer
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
diff --git a/sysui/desktop/menus/xsltfilter.desktop b/sysui/desktop/menus/xsltfilter.desktop
index 3dcad81..adea82a 100644
--- a/sysui/desktop/menus/xsltfilter.desktop
+++ b/sysui/desktop/menus/xsltfilter.desktop
@@ -9,4 +9,4 @@ GenericName=XSLT based filters
 NoDisplay=true
 StartupNotify=true
 X-GIO-NoFuse=true
-X-KDE-Protocols=file,http,smb,ftp,webdav
+X-KDE-Protocols=file,http,ftp,webdav
-- 
1.9.3

